templates/videos.html

```diff
@@ -172,46 +172,33 @@ <h4 class="u-align--center">Try adjusting your filter criteria</h4>
     const resultsCounter = document.getElementById('filter-results-count');
     const paginationContainer = document.getElementById('pagination-container');
     
-    // Create mappings between IDs and slugs
-    const idToSlug = {
-      // For topics
-      {% for tag in topic_tags %}
-      "{{ tag.id }}": "{{ tag.name|slugify }}",
-      {% endfor %}
-      // For events
-      {% for tag in event_tags %}
-      "{{ tag.id }}": "{{ tag.name|slugify }}",
-      {% endfor %}
-      // For dates
-      {% for tag in date_tags %}
-      "{{ tag.id }}": "{{ tag.name|slugify }}",
-      {% endfor %}
-      // For presenters
-      {% for presenter in presenters %}
-      "{{ presenter.id }}": "{{ presenter.name|slugify }}",
-      {% endfor %}
+    {% set filter_types = [
+        {'key': 'topic', 'items': topic_tags, 'label': 'name'},
+        {'key': 'event', 'items': event_tags, 'label': 'name'},
+        {'key': 'date', 'items': date_tags, 'label': 'name'},
+        {'key': 'presenter', 'items': presenters, 'label': 'name'}
+    ] %}
+
+    {% for filter in filter_types %}
```

Comment:
This mixture of Jinja and JS makes it very hard to parse. You need to figure out the result of Jinja that is a JS code, and then realise what JS code does.

I wander if we could simplify it. Maybe we could do the data preparation on the Python side (not in template), and then pass a whole dictionary as JSON to JS and take it from there, without the confusing looping in the template.

But maybe that's refactoring for another PR. This seems to work as expected so far.

Reply:
Reported as https://warthogs.atlassian.net/browse/WD-29981 to be handled separately.